face/UnifiedFaceCollection/python/unified_face_collection.py

```diff
@@ -1,117 +1,131 @@
-import requests
-import time
 import json
-from shared_functions import detect_faces, enlarge_bounding_box, get_image_dimensions
+from PIL import Image
+from azure.core.credentials import AzureKeyCredential
+from azure.ai.vision.face import FaceClient, FaceAdministrationClient
+from azure.ai.vision.face.models import FaceDetectionModel, FaceRecognitionModel
 
 class UnifiedFaceCollection:
     def __init__(self, subscription_key, endpoint, face_collection_id, injection_header):
-        self.subscription_key = subscription_key
-        self.endpoint = endpoint
-        self.face_api_url = f"{self.endpoint}/face/v1.0"
+        self.face_client = FaceClient(endpoint, AzureKeyCredential(subscription_key), headers = {"X-MS-AZSDK-Telemetry": injection_header})
+        self.face_admin_client = FaceAdministrationClient(endpoint, AzureKeyCredential(subscription_key), headers = {"X-MS-AZSDK-Telemetry": injection_header})
         self.face_collection_id = face_collection_id
         self.large_face_list_id = face_collection_id + "_face_list"
         self.large_person_group_id = face_collection_id + "_person_group"
-        self.injection_header = injection_header
-        self.headers = {
-            'Ocp-Apim-Subscription-Key': self.subscription_key,
-            'Content-Type': 'application/json',
-            'X-MS-AZSDK-Telemetry': injection_header
-        }
         self.create_collections()
 
+    def detect_faces(self, image_path):
+        with open(image_path, 'rb') as image_data:
+            detected_faces = self.face_client.detect(
+                    image_content=image_data.read(),
+                    detection_model=FaceDetectionModel.DETECTION03,
+                    recognition_model=FaceRecognitionModel.RECOGNITION04,
+                    return_face_id=True,
+            )
+        return detected_faces
+
+    def enlarge_bounding_box(self, face_rectangle, image_width, image_height, enlargement_factor=1.2):
+        left = max(0, face_rectangle['left'] - (face_rectangle['width'] * (enlargement_factor - 1) / 2))
+        top = max(0, face_rectangle['top'] - (face_rectangle['height'] * (enlargement_factor - 1) / 2))
+        width = min(image_width - left, face_rectangle['width'] * enlargement_factor)
+        height = min(image_height - top, face_rectangle['height'] * enlargement_factor)
```

Comment:
FaceRectangle has built-in attributes (https://learn.microsoft.com/en-us/python/api/azure-ai-vision-face/azure.ai.vision.face.models.facerectangle?view=azure-python-preview#variables) to access the top/left/width/height values:

```suggestion
        left = max(0, face_rectangle.left - (face_rectangle.width * (enlargement_factor - 1) / 2))
        top = max(0, face_rectangle.top - (face_rectangle.height * (enlargement_factor - 1) / 2))
        width = min(image_width - left, face_rectangle.width * enlargement_factor)
        height = min(image_height - top, face_rectangle.height * enlargement_factor)
```